scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,238 @@
+"""
+A VIVO harvester for the SHARE project
+
+This harvester makes a SPARQL query to the VIVO SPARQL endpoint,
+the information to acces to a VIVO endpoint must be provided in the local.py file.
+"""
+
+from __future__ import unicode_literals
+
+import json
+import logging
+
+from datetime import date, timedelta, datetime
+
+from six.moves import xrange
+from SPARQLWrapper import SPARQLWrapper, JSON
+
+from scrapi import settings
+from scrapi.settings.sparql_mapping import *
+from scrapi.base import JSONHarvester
+from scrapi.linter.document import RawDocument
+from scrapi.base.helpers import build_properties
+
+logger = logging.getLogger(__name__)
+
+
+def process_object_uris(pmid, doi):
+    ret = []
+    if pmid:
+        pmid = 'http://www.ncbi.nlm.nih.gov/pubmed/{}'.format(pmid)
+        ret.append(pmid)
+    if doi:
+        doi = 'https://dx.doi.org/{}'.format(doi)
+        ret.append(doi)
+    return ret
+
+
+class VIVOHarvester(JSONHarvester):
+    short_name = 'vivo'
+    long_name = 'VIVO'
+    url = settings.VIVO_ACCESS['url']
+    base_url = settings.VIVO_ACCESS['query_endpoint']
+    sparql_wrapper = SPARQLWrapper(base_url)
+    sparql_wrapper.setReturnFormat(JSON)
+    sparql_wrapper.addParameter("email", settings.VIVO_ACCESS['username'])
+    sparql_wrapper.addParameter("password", settings.VIVO_ACCESS['password'])
+    sparql_wrapper.method = 'GET'
+
+    DEFAULT_ENCODING = 'UTF-8'
+    QUERY_TEMPLATE = """
+                       PREFIX vivo:  <http://vivoweb.org/ontology/core#>
+                       PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+                       PREFIX rdfs:  <http://www.w3.org/2000/01/rdf-schema#>
+                       PREFIX bibo:  <http://purl.org/ontology/bibo/>
+                       PREFIX foaf:  <http://xmlns.com/foaf/0.1/>
+                       PREFIX vitro: <http://vitro.mannlib.cornell.edu/ns/vitro/0.7#>
+                       PREFIX dc: <http://purl.org/dc/terms/>
+                       PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
+                       PREFIX obo:   <http://purl.obolibrary.org/obo/>
+
+                       SELECT {}
+                       {{
+                            {}
+                        }}
+                     """
+
+    record_encoding = None
+
+    def get_string(self, uri, sparql_map):
+        variable = sparql_map['name']
+        pattern = sparql_map['pattern'].format(uri)
+        sparql_query = self.QUERY_TEMPLATE.format('?' + variable, pattern)
+        self.sparql_wrapper.setQuery(sparql_query)
+        result = self.sparql_wrapper.query()
+        result = result.convert()
+        if len(result['results']['bindings']) > 0:
```

Comment:
I think just checking `if result['results']['bindings']:` will work as well here, and is a little clearer.